addons/point_of_sale/models/res_partner.py

```diff
@@ -36,7 +36,8 @@ def create_from_ui(self, partner):
         if partner_id:  # Modifying existing partner
             self.browse(partner_id).write(partner)
         else:
-            partner['lang'] = self.env.user.lang
+            if not partner['lang']:
```

Comment:
I don't think you need this, because the default value of lanf on python field is 'default=lambda self: self.env.lang,' So I think you can remove this 'if not lang'